sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -544,6 +603,41 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
     result
   }
 
+  private def loadMap(startVersion: Long, endVersion: Long): HDFSBackedStateStoreMap = {
```

Comment:
Same, ideally the semantic (intention) needs to be explained in either method name or param name. This leads confusion for the behavior.

Reply:
Once the intention and usage is clearly separated to state data source reader (one time read), we probably won't prefer using the cache. Since the provider is bound to the state data source reader, it wouldn't matter whether to use the cache or not, but using cache would give unnecessary question/concern/doubt.

For example, snapshot starting version must read from snapshot file, but cache does not have such semantic, so when you get from cache we should check whether the cache is only updated from reading snapshot.